src/PointwiseFunctions/GeneralRelativity/WeylMagnetic.cpp

```diff
@@ -0,0 +1,139 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/GeneralRelativity/WeylMagnetic.hpp"
+
+#include "DataStructures/LeviCivitaIterator.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/VectorImpl.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+///\cond
+namespace gr {
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::ii<DataType, SpatialDim, Frame> weyl_magnetic(
+    const tnsr::ijj<DataType, SpatialDim, Frame>& grad_extrinsic_curvature,
+    const tnsr::ii<DataType, SpatialDim, Frame>& spatial_metric) noexcept {
+  tnsr::ii<DataType, SpatialDim, Frame> weyl_magnetic_part{};
+  weyl_magnetic<SpatialDim, Frame, DataType>(make_not_null(&weyl_magnetic_part),
+                                             grad_extrinsic_curvature,
+                                             spatial_metric);
+
+  return weyl_magnetic_part;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void weyl_magnetic(
+    const gsl::not_null<tnsr::ii<DataType, SpatialDim, Frame>*>
+        weyl_magnetic_part,
+    const tnsr::ijj<DataType, SpatialDim, Frame>& grad_extrinsic_curvature,
+    const tnsr::ii<DataType, SpatialDim, Frame>& spatial_metric) noexcept {
+  auto grad_extrinsic_curvature_cross_spatial_metric =
+      make_with_value<tnsr::ij<DataType, SpatialDim, Frame>>(
+          get<0, 0>(spatial_metric), 0.0);
+
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    for (size_t j = 0; j < SpatialDim; ++j) {
+      for (LeviCivitaIterator<3> it; it; ++it) {
+        grad_extrinsic_curvature_cross_spatial_metric.get(i, j) +=
+            it.sign() * grad_extrinsic_curvature.get(it[2], it[1], i) *
+            spatial_metric.get(j, it[0]);
+      }
+    }
+  }
```

Comment:
I think you can probably just omit this temporary and assign the components directly to `weyl_magnetic_part` in the first loop.
I suggest something like (if it works):
```
for (size_t i = 0; i < SpatialDim; ++i) {
  for (size_t j = 0; j < SpatialDim; ++j) {
    LeviCevitaIterator<3> it;
    if (i <= j) {
      weyl_magnetic_part->get(i, j) = (i != j ? 0.5 : 1.0) * it.sign() *
         grad_extrinsic_curvature.get(it[2], it[1], i) * 
         spatial_metric.get(j, it[0]);
         ++it;
    }
    for (; it; ++it) {
       weyl_magnetic_part->get(i, j) += (i != j ? 0.5 : 1.0) * it.sign() *
         grad_extrinsic_curvature.get(it[2], it[1], i) * 
         spatial_metric.get(j, it[0]);
    }
    if (i >= j) {
      weyl_magnetic_part->get(i, j) /= sqrt_det_spatial_metric;
    }
  }
}
```
I also suggest that you should take the sqrt_det_spatial_metric as an argument to this function -- it's needed pretty frequently so it's usually in the databox, and that would be more efficient that computing it in the loops here.

Reply:
I tried this code and the test failed. Would you like to meet up and figure out what the problem is or should I convert back to the old code?

Reply:
Hm, okay, no worries then -- you can leave it as it was with the temporary so that we can move forward with the PR. Can you take the `sqrt_det_spatial_metric` as a function argument, though?